File: code/Java/interpreter/expr/NotBoolExpr.java
package interpreter.expr;

public class NotBoolExpr extends BoolExpr {

    private BoolExpr expr;

    public NotBoolExpr(int line, BoolExpr expr) {
        super(line);
        this.expr = expr;
    }

    public boolean expr() {
        return !expr.expr();
    }   

}

File: code/C++/interpreter/expr/NotBoolExpr.cpp
#include "NotBoolExpr.h"

NotBoolExpr::NotBoolExpr(int line, BoolExpr* expr)
	: BoolExpr(line), m_expr(expr) {
}

NotBoolExpr::~NotBoolExpr() {
	delete m_expr;
}

bool NotBoolExpr::expr() {
	return !m_expr->expr();
}
